src/components/ListItems.jsx

```diff
@@ -0,0 +1,309 @@
+import React, { useState, useRef, useEffect } from "react";
+import { useDispatch } from "react-redux";
+
+// @material-ui import
+import IconButton from "@material-ui/core/IconButton";
+import DeleteIcon from "@material-ui/icons/Delete";
+import AddCircleRoundedIcon from "@material-ui/icons/AddCircleRounded";
+import RemoveCircleIcon from "@material-ui/icons/RemoveCircle";
+
+// actions import
+import { addItem, deleteItem } from "../actions/ItemActions";
+import {
+    addSelectedTopicsItem,
+    deleteSelectedTopicsItem,
+} from "../actions/selectedTopicsItemActions";
+
+import "../App.css";
+
+export const ListItems = (props) => {
+    // Initial setup
+    const [btnAddNewItemClicked, setBtnAddNewItemClicked] = useState(false);
+    const [inputEditedItemVisible, setInputEditedItemVisible] = useState(false);
+    const [inputNewItem, setInputNewItem] = useState("");
+    const [editedItem, setEditedItem] = useState(null);
+    const [editedItemIndex, setEditedItemIndex] = useState(null);
+    const [inputEditedItem, setInputEditedItem] = useState("");
```

Comment:
When handling multiple, interdependent state cells you should consider `useReducer` as an alternative

Reply:
Even if it's just a local state?